milatools/datasets/torch.py

```diff
@@ -0,0 +1,66 @@
+import os
+import subprocess
+from typing import Tuple
+
+import torchvision.datasets
+
+import milatools.utils
+
+DATASET_PATH = "/network/datasets/{}.var/{}_torchvision"
+
+
+def fetch_imagenet(local_directory=None):
+    if milatools.running_on_mila_cluster:
+        dataset_home = os.path.join(os.environ["SLURM_TMPDIR"], "ImageNet")
+        train_directory = os.path.join(dataset_home, "train")
+        validation_directory = os.path.join(dataset_home, "val")
+
+        subprocess.run(f"mkdir -p {train_directory}/ {validation_directory}/", shell=True)
+        subprocess.run(f"tar -xf /network/datasets/imagenet/ILSVRC2012_img_train.tar -C {train_directory}", shell=True)
```

Comment:
One thing I'm pondering is using this library in milatools instead of subprocess: https://plumbum.readthedocs.io/en/latest/

It looks nicer and it can run commands over SSH, and it seems to be pretty light. Thoughts?

Reply:
It seems more overhead to learn the abstractions/interface than what it saves, we could just do something like:
```python
local = lambda cmd: subprocess.run(cmd, shell=True)
remote = fabric.Connection(host="somehost").run
```
no?
(not that  i think it's a good idea to do the above)

Reply:
I don't know, the interface seems fairly intuitive and I reopen the `subprocess` docs nearly every time I use it anyway. I assume it also does proper escaping, which makes it less of a footgun than `shell=True` (if e.g. train_directory contains a space or special character). Either way, I have yet to try it.

Reply:
I have tried a platform independent version that does away with the shell=True but the decompression relies on the shell quite heavily (see: https://github.com/mila-iqia/milatools/blob/35676c3bdde67188e00dbb550d9e170081d8d383/milatools/datasets/torch.py#L22), how can we replace that?

Reply:
I have tried a platform independent version that does away with the shell=True but the decompression relies on the shell quite heavily (see: https://github.com/mila-iqia/milatools/blob/35676c3bdde67188e00dbb550d9e170081d8d383/milatools/datasets/torch.py#L22), how can we replace that?

Reply:
I have tried a platform independent version that does away with the shell=True but the decompression relies on the shell quite heavily (see: https://github.com/mila-iqia/milatools/blob/35676c3bdde67188e00dbb550d9e170081d8d383/milatools/datasets/torch.py#L22), how can we replace that?

Reply:
I think I would just use pure Python, e.g. a combination of `os.walk` and `shutil.unpack_archive`, although I'm not sure about the performance characteristics.